### PR TITLE
[debugger] Fix suspend_policy that will be send to debugger-libs

### PR DESCRIPTION
**Pull request by thaystg**
Fix suspend_policy that will be send to debugger-libs, because on debugger-libs we want to resume the VM only when it's suspended, but we were sending the wrong suspend_policy in the case of EVENT_KIND_VM_START and mono is started with suspend=y.
With this fix we can do this PR again: https://github.com/mono/debugger-libs/pull/264/

Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/999375/

Related https://github.com/mono/debugger-libs/pull/302

### CODE REVIEW

File: mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/Connection.cs [hunk 1]
```diff
@@ -799,7 +799,7 @@ class PacketReader {
 				ReadInt (); // id
 				ReadByte (); // flags
 				ErrorCode = ReadShort ();
-				if (ErrorCode == (int)Mono.Debugger.Soft.ErrorCode.INVALID_ARGUMENT && len > offset)
+				if (ErrorCode == (int)Mono.Debugger.Soft.ErrorCode.INVALID_ARGUMENT && connection.Version.AtLeast (2, 55) && len > offset)
 					ErrorMsg = ReadString ();
```
Comment (vargaz) [inline mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/Connection.cs]: Unrelated changes.

File: mono/mini/debugger-agent.c [hunk 2]
```diff
@@ -6961,6 +6963,11 @@ get_types_for_source_file (gpointer key, gpointer value, gpointer user_data)
 	}
 }
 
+static void add_error_string (Buffer *buf, const char *str) {
+	if (CHECK_PROTOCOL_VERSION (2, 55)) 
+		buffer_add_string (buf, str);
```
Comment (vargaz) [inline mono/mini/debugger-agent.c]: Ditto.

File: mono/mini/debugger-agent.c [hunk 3]
```diff
@@ -7265,7 +7272,7 @@ vm_commands (int command, int id, guint8 *p, guint8 *end, Buffer *buf)
 		ignore_case = decode_byte (p, &p, end);
 
 		if (!mono_reflection_parse_type_checked (name, &info, error)) {
-			buffer_add_string (buf, mono_error_get_message (error));
+			add_error_string (buf, mono_error_get_message (error));
```
Comment (vargaz) [inline mono/mini/debugger-agent.c]: Ditto.

**Comment by thaystg** [comment]
@monojenkins build failed

**Comment by thaystg** [comment]
@monojenkins backport 2020-02
